### ISSUE TITLE
оптимизация записей в deps

### ISSUE THREAD

**Issue by kompolom** [bem/bem-forum-content-ru#527]
Сейчас подключаю клиентские шаблоны так:

````
{
  tech: 'js',
  sholdDeps: [
     {
        block: 'modal',
        tech: 'bemhtml'
     }
  ]
}
````

Хотелось бы не писать `tech: bemhtml` каждому блоку, а  написать один раз. Это возможно?

**Comment by tadatuta** [comment]
@kompolom Пока ничего лучше, чем воспользоваться тем фактом, что `deps.js` — это просто JS-код и заиспользовать там какие-нибудь функции, предложить не могу :(
```js
({
    tech: 'js',
    shouldDeps: [
        {
            block: 'b1'
        },
        'b2',
        {
            block: 'b3',
            elem: 'e1'
        }
    ].map(function(item) {
        if (typeof item === 'string') {
            item = {
                block: item
            };
        }

        item.tech = 'bemhtml';

        return item;
    })
})
```

**Comment by zxqfox** [comment]
А разве это вообще нужно писать?

/cc @blond

**Comment by blond** [comment]
@zxqfox А почему считаешь, что не нужно? :)

Говорится буквально следующее «JS-реализация блока `this.block` зависит от BEMHTML-реализации блока `modal`».

**Comment by zxqfox** [comment]
В мечтах: я вижу депс без технологий, где блоки сами технологии свои экспортируют. Волшебный лес, какой-то. Ок, простите.

**Comment by tadatuta** [comment]
@zxqfox если допустить, что мечта сбылась, где-то все равно нужно будет сказать, что из всех экспортируемых реализаций блока ты хочешь именно BEMHTML (а не BH, например).

**Comment by zxqfox** [comment]
Да, где-то в импортируемом блоке. Пока не знаю как именно, но явно не в сборке, и не в блоке-импортере.
